### PR TITLE
Jenkins pipeline

### PR DESCRIPTION
**Pull request by glogiotatidis**
Full Jenkins pipeline to

 - Build Docker Images
 - Test
 - Push Docker Images to Registry
 - Conditionally (`onBranch`) Push to Stage
 - Conditionally (`onTag`) and `input` push to usw and then again with `input` to euw

Sends IRC notifications when complete

Needs the following Jenkins permissions
```
method hudson.model.Run getPreviousBuild
method hudson.model.Run getResult
method org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper getRawBuild
staticField hudson.model.Result SUCCESS
staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods split java.lang.String
```

@jgmize lets discuss this approach tomorrow, especially the permissions

See also https://github.com/mozmar/jenkins-pipeline/pull/2

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,129 @@
+@Library('github.com/glogiotatidis/jenkins-pipeline@more-commands')
```
Comment (jgmize) [inline Jenkinsfile]: As discussed, once https://github.com/mozmar/jenkins-pipeline/pull/2 is finished and merged please s/glogiotatidis/mozmar/ and then update the perms on the jenkins job, and then we should be good to merge this PR.

**Comment by glogiotatidis** [comment]
Demo of this pipeline can be found here https://ci.us-west.moz.works/blue/organizations/jenkins/CareersPipelines/detail/master/4/pipeline

File: Jenkinsfile (hunk 1 above)
Comment (jgmize) [inline Jenkinsfile]: Just merged https://github.com/mozmar/jenkins-pipeline/pull/2 so you should be able to update this to use it now.

Comment (glogiotatidis) [inline Jenkinsfile]: Updated!

**Pull request merged**